src/h2olog/h2olog.cc

```diff
@@ -421,6 +419,18 @@ int main(int argc, char **argv)
             expr += ')';
         }
         cflags.push_back(build_cc_macro_expr("H2OLOG_IS_SAMPLING_ADDRESS(family, addr)", std::string("(") + expr + ")"));
+        selective_tracing = true;
+    }
+
+    if (selective_tracing) {
+        cflags.push_back(build_cc_macro_expr("H2OLOG_SELECTIVE_TRACING", 1));
+    }
+
+    for (const auto &usdt : tracer->usdt_probes()) {
+        if (!selective_tracing && usdt.fully_qualified_name() == "h2o:_private_socket_lookup_flags")
```

Comment:
Instead, would it be possible to add `_private_socket_lookup_flags` when it should be added?

The reason I ask is at the moment, it is easy for users to break selective tracing. If the user uses `-t` but does not explicitly name `_private_socket_lookup_flags`, then things would fail.

Reply:
Good point. Fixed in 715d3dfc7ac410ad5e9c3b615c23c9c78f134ccc and now `_private_socket_lookup_flags` is independent from `usdt_probes()` and/or the `-t` option.